packages/discovery-react-components/src/components/StructuredQuery/components/RuleRow/RuleRow.tsx

```diff
@@ -45,15 +54,20 @@ export const RuleRow: FC<RuleRowProps> = ({
   const showRemoveRuleRowButton =
     structuredQuerySelection.groups[groupId].rows.length > 1 || !isTopLevelGroup;
 
+  // TODO Check out missing prop warning on <ComboBox />
   return (
     <div className={structuredQueryRulesClass} data-testid={`rule-row-${groupId}`}>
       <ComboBox
         id={`structured-query-rules-field-${groupId}`}
-        // TODO: Items is empty for now as it's a required field and retrieving fields for the dropdown
-        // and adding them as items will be addressed in a future issue
-        items={[]}
-        placeholder={messages.fieldDropdownPlaceholderText}
+        items={projectFields}
+        placeholder={
+          haveProjectFields
+            ? messages.fieldDropdownPlaceholderText
+            : messages.fieldDropdownLoadingText
+        }
         titleText={messages.fieldDropdownTitleText}
+        data-test-id={`structured-query-rules-field-${groupId}`}
```

Comment:
i don't see this being used in tests, but if it were, i would imagine that it should be `data-testid` instead

Reply:
I've removed this id that's no longer needed